### ISSUE TITLE
Issue with editing a factorial 

### ISSUE THREAD

**Issue by Janewanguimugo** [femiguez/apsimx#104]
When I inspect my the weather fiIe I get the correct value
inspect_apsimx("soy_fac_jane.apsimx", src.dir = extd.dir,
               root = c("pd", "Base"),
               node = "Weather")

Met file: Angonia.met

When I want to edit the weather node and replace it with another met file I get an error that the weather node cannot be found
edit_apsimx("soy_fac_jane.apsimx", 
            src.dir = extd.dir, 
            node = "Weather",
            overwrite = TRUE,
            root = c("pd", "Base"), 
            value = "chitedze.met")

Error: Weather node not found

The factorial looks like this
![image](https://user-images.githubusercontent.com/103818039/199172506-2980004e-21b8-4832-afef-cbcaa1222445.png)

**Comment by femiguez** [comment]
Which version of 'apsimx' are you running? The current version in github seems to work correctly. For example,

```
library(apsimx)

tmpd <- tempdir()
setwd(tmpd)

ex.dir <- auto_detect_apsimx_examples()

file.copy(file.path(ex.dir, "Factorial.apsimx"), ".")

inspect_apsimx("Factorial.apsimx",
               node = "Weather",
               root = c("Permutation", "Base4"))
## Met file: %root%\Examples\WeatherFiles\lincoln.met 

edit_apsimx("Factorial.apsimx",
            node = "Weather",
            root = c("Permutation", "Base4"),
            value = "Ames.met")
## Edited (node):  Weather 
## Edited (child):  none 
## Edited parameters:  
## New values:  Ames.met 
## Created:  ./Factorial-edited.apsimx

inspect_apsimx("Factorial-edited.apsimx",
                node = "Weather",
                root = c("Permutation", "Base4"))
## Met file: Ames.met 

```

**Comment by Janewanguimugo** [comment]
My experiment only has one base. I have attached the apsimx file. I am
using the most up-to-date version of APSIMx. Please see attached file.



Regards
Jane Wangui Mugo (PhD)
Consultant Agriculture and Meteorology
______________________________
Website: https://www.linkedin.com/in/jane-wangui-mugo-43b73488/
Mobile: 0713003792

**Comment by Janewanguimugo** [comment]
[soy_fac_jane.zip](https://github.com/femiguez/apsimx/files/9910265/soy_fac_jane.zip)
Here it is attached

**Comment by Janewanguimugo** [comment]
I have installed the newest version 2.3.9 but the problem still persists. I
do not get any errors with the factorial example that comes with APSIMx.
The problem is when you only have one experiment with one base as opposed
to 4 experiments with 4 bases.

**Comment by Janewanguimugo** [comment]
How to replace the soil profile ...I get the error below
 edit_apsimx_replace_soil_profile(crop, root = c("pd", "Base"), soil.profile = sp, overwrite = TRUE)

Error in edit_apsimx_replace_soil_profile(crop, root = c("pd", "Base"),  : 
  object 'wcore1' not found

**Comment by Janewanguimugo** [comment]
crop is "soy_fac_jane.apsimx"

**Comment by femiguez** [comment]
It worked for you (the previous issue)? I forgot to mention that I modified the file you sent me a bit. 
[soy_fact_jane_fem.apsimx.zip](https://github.com/femiguez/apsimx/files/9941328/soy_fact_jane_fem.apsimx.zip)

Then, this works
```
 library(apsimx) 
inspect_apsimx("soy_fact_jane_fem.apsimx",
               node = "Weather",
               root = c("pd", "Base_one"))

edit_apsimx("soy_fact_jane_fem.apsimx",
            node = "Weather",
            root = c("pd", "Base_one"),
            value = "chitedze.met")

inspect_apsimx("soy_fact_jane_fem-edited.apsimx",
               node = "Weather",
               root = c("pd", "Base_one")) 
```

**Comment by femiguez** [comment]
I will take a look at the soil profile issue

**Comment by Janewanguimugo** [comment]
In the previous issue it worked, except when it came to editing the soil profile.

**Comment by Janewanguimugo** [comment]
Question...for the factorial to work must it have more than one base?

**Comment by femiguez** [comment]
@Janewanguimugo some of these functions not working when there is a factorial and only one simulation was not my intention. I just pushed one commit that seems to solve the problem. The example I tested is the one I sent you with two simulations. Will be testing this more in the future. Thanks for trying the package and the feedback!

**Comment by Janewanguimugo** [comment]
Thank you I will work with that recommendation.

**Comment by Janewanguimugo** [comment]
#Source of soil data https://soilgrids.org/
sp <- get_isric_soil_profile(lonlat = c(33.64, -13.98))

sp <- apsimx_soil_profile()
extd.dir <- system.file("extdata", package = "apsimx")

## I write to a temp directory but replace as needed
tmp.dir <- tempdir()

edit_apsimx_replace_soil_profile("Maize.apsimx", soil.profile = sp,
                                 src.dir = extd.dir, wrt.dir = tmp.dir)



Error in edit_apsimx_replace_soil_profile("Maize.apsimx", soil.profile =
sp,  :
  argument "root" is missing, with no default
>







Regards
Jane Wangui Mugo (PhD)
Consultant Agriculture and Meteorology
______________________________
Website: https://www.linkedin.com/in/jane-wangui-mugo-43b73488/
Mobile: 0713003792

**Comment by femiguez** [comment]
Thanks again for the report! I tried to fix this issue with version 2.4.6.

I will be testing this a bit more
